### PR TITLE
Nexus read

### PR DESCRIPTION
**Pull request by RubelMozumder**
@shabihsherjeel, After you are done with your code you can merge this branch.

### CODE REVIEW

**Comment by shabihsherjeel** [comment]
@RubelMozumder I have removed my changes as there are some readers that need to be fixed with it. I'll make another branch for that. My changes aren't a blocker for this MR anymore.

File: nexusutils/nexus/nexus.py [hunk 1]
```diff
@@ -1104,32 +1126,109 @@ def axis_helper(dim, nxdata, signal, axes, logger):
...
+    required=False,
+    default=None,
+    help="To view documentation. E.g. -d /entry/data/delays"
+)
+@click.option(
+    '-c',
+    '--concept',
+    required=False,
+    default=None,
+    help="To view concept. E.g. -d /entry/data/delays"
```
Comment (shabihsherjeel) [inline nexusutils/nexus/nexus.py:1229]: The flag in the example here is wrong. It should be `-c`.

File: nexusutils/nexus/nexus.py [hunk 2]
```diff
@@ -1138,9 +1237,14 @@ def main():
     logger = logging.getLogger(__name__)
     logger.addHandler(stdout_handler)
     logger.setLevel(logging.DEBUG)
-    nexus_helper = HandleNexus(logger, sys.argv[1:])
+    if documentation and concept:
+        raise ValueError("Only documentation (-d) or is_a relation with a concept (-c)"
```
Comment (shabihsherjeel) [inline nexusutils/nexus/nexus.py:1241]: Consider adding `Only one of` or something of the sorts. Just the `or` does not seem sufficient. I guess the idea is to tell users that they can only have either -d or -c but not both.

**Review by shabihsherjeel — CHANGES_REQUESTED** [review]

File: nexusutils/nexus/README.md [hunk 3]
```diff
@@ -2,8 +2,20 @@ This utility outputs a debug log for a given NeXus file by annotating the data a
 metadata entries with the schema definitions from the respective NeXus base classes
 and application definitions to which the file refers to.
 
-```console
-user@box:~$ read_nexus <path_to_nexus_file>
+```
+Options:
+    --nexus-file : Name of nexus output file (.nxs) to enquiry.
```
Comment (sanbrock) [inline nexusutils/nexus/README.md:7]: please make this option also available as -f  
and also take the argument with no option specification as the nexus-file. Documentation should read:
read_nexus [Options] <path_to_nexus_file>

File: nexusutils/nexus/README.md [hunk 4]
```diff
@@ -2,8 +2,20 @@ This utility outputs a debug log for a given NeXus file by annotating the data a
...
+    -d, --documentation : Definition path in nexus output (.nxs) file. Returns debug
+                          log relavent with that definition path.
```
Comment (sanbrock) [inline nexusutils/nexus/README.md:9]: add example from the default neXus file: (like /entry/definition)

File: nexusutils/nexus/README.md [hunk 5]
```diff
@@ -2,8 +2,20 @@ This utility outputs a debug log for a given NeXus file by annotating the data a
...
+    -c, --concept : Concept path from application definition file (.nxdl,xml). Finds out
+                    all the available concept definition (IS-A realation) for rendered
+                    concept path.
```
Comment (sanbrock) [inline nexusutils/nexus/README.md:12]: also add example here: (like /NXentry/definition)

File: nexusutils/nexus/nexus.py [hunk 6]
```diff
@@ -1104,32 +1126,109 @@ def axis_helper(dim, nxdata, signal, axes, logger):
...
+        # Aggregating hdf path corresponds to concept query node
+        self.hdf_path_list_for_c_inq_nd = []
 
     def visit_node(self, hdf_name, hdf_node):
         """Function called by h5py that iterates on each node of hdf5file.
         It allows h5py visititems function to visit nodes."""
-        hdf_path = '/' + hdf_name
-        process_node(hdf_node, hdf_path, self.parser, self.logger)
+        if self.d_inq_nd is None and self.c_inq_nd is None:
+            process_node(hdf_node, '/' + hdf_name, self.parser, self.logger)
```
Comment (sanbrock) [inline nexusutils/nexus/nexus.py:1186]: when run from command line, can we set a single logger, so each line is printed only once?

File: tests/data/nexus/Ref1_c_option_test.log [hunk 7]
```diff
@@ -0,0 +1,4 @@
+INFO: entry/instrument/beam_probe_0
+INFO: entry/instrument/beam_probe_0
```
Comment (sanbrock) [inline tests/data/nexus/Ref1_c_option_test.log:2]: we should then have no duplicated lines here

File: tests/data/nexus/Ref_nexus_test.log [hunk 8]
```diff
@@ -296,7 +296,7 @@ DEBUG - classes:
 NXdata.nxdl.xml:/AXISNAME
 DEBUG - @units - REQUIRED, but undefined unit category
 DEBUG - ===== FIELD (//entry/data/data): <HDF5 dataset "data": shape (80, 146, 195), type "<f4">
-DEBUG - value: [[0.0000000e+00 0.0000000e+00 0.0000000e+00 0.0000000e+00 0.0000000e+00 ...
+DEBUG - value: [[0. 0. 0. ... 0. 0. 0.] ...
```
Comment (sanbrock) [inline tests/data/nexus/Ref_nexus_test.log:299]: I remember, it was a CI/CD problem with this (probably different version of the library printing)

**Review by sanbrock — CHANGES_REQUESTED** [review]

File: README.md [hunk 9]
```diff
@@ -39,9 +39,9 @@ data into the NeXus standard and visualising the files content.
 
 # Command line tools
 
-- [**nyaml2nxdl**](https://github.com/nomad-coe/nomad-parser-nexus/blob/master/nexusutils/nyaml2nxdl/README.md): Converts, easy to read, YAML [NeXus schemas](https://nexusformat.org) into NeXus XML definition language (NXDL).
-- [**dataconverter**](https://github.com/nomad-coe/nomad-parser-nexus/blob/master/nexusutils/dataconverter/README.md): Creates compliant instances of NeXus/HDF5 files to [NeXus schemas](https://nexusformat.org).
-- [**read_nexus**](https://github.com/nomad-coe/nomad-parser-nexus/blob/master/nexusutils/nexus/README.md): Outputs a debug log for a given NeXus file.
+- [**nyaml2nxdl**](nexusutils/nyaml2nxdl/README.md): Converts, easy to read, YAML [NeXus schemas](https://nexusformat.org) into NeXus XML definition language (NXDL).
+- [**dataconverter**](nexusutils/dataconverter/README.md): Creates compliant instances of NeXus/HDF5 files to [NeXus schemas](https://nexusformat.org).
+- [**read_nexus**](nexusutils/nexus/README.md): Outputs a debug log for a given NeXus file.
```
[the remark this answers is not in the document]
Comment (shabihsherjeel) [reply README.md:44]: The problem is that this Readme gets rendered on the PyPi webpage. That requires an absolute link. So don't worry about the Readme link pointing to the master branch. Eventually that's where it should end up going. I believe there is no other way to get this resolved but to have absolute links. 

@domna you changed them to absolute for the same reasons I mentioned or is there something else too?

File: tests/data/nexus/Ref_nexus_test.log (hunk 8 above)
Comment (RubelMozumder) [reply tests/data/nexus/Ref_nexus_test.log:299]: At this moment the generated nexus_test.log is just generating '0' value without fraction part.

File: README.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (RubelMozumder) [reply README.md]: @shabihsherjeel I have removed the changes.

**Comment by sanbrock** [comment]
As I see, the forced-pushes from Rubel took away the modifications from Sherjeel and me, so I will bring those changes back.

**Review by sanbrock — APPROVED** [review]
LGTM

**Pull request merged**